gym/lib/gym/runner.rb

```diff
@@ -72,6 +73,11 @@ def clear_old_files
       end
     end
 
+    def fix_archive
+      return unless Gym.config[:use_legacy_build_api]
```

Comment:
Let's do as @samrobbins suggests and put this behavior behind a feature switch for now.

Something like:
`return unless Gym.config[:use_legacy_build_api] && !ENV['FASTLANE_GYM_USE_GENERIC_ARCHIVE_FIX'].nil?`